docs/adr/0008-preview-deployment.md

```diff
@@ -0,0 +1,80 @@
+
+Pull requests can be categorized into several types:
+
+- Dependency updates (primarily through Dependabot)
+- Feature enhancements and bug fixes that require QA verification
+- Bug fixes that do not necessitate QA action
+- Accessibility fixes that need approval from the a11y team
+
+It seems counterproductive to delay all deployments due to a small number of changesets requiring verification. Moreover, the impact of postponing a11y changes is uncertain.
+
+If we stick to our current release process, the steps would be:
+
+- Establish a release branch
+- Undo the unwanted changes (although cherry-picking the desired changes might be more appropriate in this case)
+- Tag the release branch
+- Deploy the tag to the staging environment, undoing the unwanted changes (which could interfere with the verification process)
+- Deploy to the production environment
+- Revert the staging environment to the main branch
+
+This approach appears to be disruptive and prone to errors. If our test automation and feature flags are not yet reliable enough to ensure safe continuous deployment, we should consider how to enable PR verification by QA/a11y/etc. prior to merging, similar to the process we use with Netlify and Storybook for components.
+
+We aim to identify potential solutions that would allow QA to be involved earlier in the process.
+
+In this exploration, we want to investigate how we can incorporate QA earlier in the process. A crucial step would be to explore a Heroku preview-like solution, where a self-contained environment would be set up for each PR, and QA would be able to test the changes before merging changes into the main branch.
+
+## Decision Drivers <!-- optional -->
+
+- Solution that provides the best experience for developers
+- Move away from predefined releases (weekly release cycle for stage, where QAs used to test), it's slow, manual, and incompatible with modern industry practices
+- Five keys of CICD release process
+
+## Considered Options
+
+- No Change - stagger the stage and production weekly release like we do now
+- Utilizing GCP Native tools (cloudrun + managed cloud SQL / postgres)
+- Vercel (NextJS Cloud) + Neon
+- GCP Cloud Run + Neon
+
+## Decision Outcome
+
+GCP Native only
+
+This decision was made:
+
+1. to avoid the need for additional vendor negociations
+2. simplify the architecture
+3. security (encapsulated in GCP)
+
+### Positive Consequences
+
+- Easy to test with GCP sandbox, there's no need to acquire test accounts from external vendors
+
+### Negative Consequences
+
+- The cloudSQL setup has some drawbacks compare to using serverless Postgres (Neon):
+  - The database is shared, so technically schema changes are not possible
+  - There might be a few solutions for this, none is very straightforward:
+    - namespace a new database for each PR (this process is time-consuming, expensive, and might be hard to set up in CICD)
```

Comment:
Do you think this would be that hard? It would require randomly generating a database name, creating it, and running `db:migrate` which is all pretty fast... it won't have any test data.

I'm not sure how we can keep migrations from interfering with each other (especially migrations that don't end up going further than a PR) otherwise...

Reply:
Good question, I actually am just experimenting with this today, and I think I just got it to work: https://github.com/mozilla/blurts-server/pull/4517